monai/transforms/post/array.py

```diff
@@ -122,94 +122,65 @@ class AsDiscrete(Transform):
     Args:
         argmax: whether to execute argmax function on input data before transform.
             Defaults to ``False``.
-        to_onehot: whether to convert input data into the one-hot format.
-            Defaults to ``False``.
-        num_classes: the number of classes to convert to One-Hot format.
+        to_onehot: if not None, convert input data into the one-hot format with specified number of classes.
+            Defaults to ``None``.
+        threshold_values: if not None, threshold the float values to int number 0 or 1 with specified theashold.
             Defaults to ``None``.
-        threshold_values: whether threshold the float value to int number 0 or 1.
-            Defaults to ``False``.
-        logit_thresh: the threshold value for thresholding operation..
-            Defaults to ``0.5``.
         rounding: if not None, round the data according to the specified option,
             available options: ["torchrounding"].
 
-    .. deprecated:: 0.6.0
-        ``n_classes`` is deprecated, use ``num_classes`` instead.
-
     """
 
     backend = [TransformBackends.TORCH]
 
-    @deprecated_arg("n_classes", since="0.6")
     def __init__(
         self,
         argmax: bool = False,
-        to_onehot: bool = False,
-        num_classes: Optional[int] = None,
-        threshold_values: bool = False,
```

Comment:
this will break many tutorials, we may provide better error messages
```py
@deprecated_arg(name="threshold_values", new_name="threshold", since="0.8")
def __init__(threshold = None, ...):
   if threshold_values is True:
       raise ValueError("threshold_values=True is deprecated, please use threashold=value instead.")
```

Reply:
Sure, let me add the error message.

Thanks.

Reply:
had a typo `since="0.8"` should be 0.7, otherwise it won't work with the release candidates such as v0.8rc0